File: Assets/zfoocs/Gateway/GatewayToProviderResponse.cs
using System;
using System.Collections.Generic;

namespace zfoocs
{
    
    public class GatewayToProviderResponse
    {
        public string message;

        public static GatewayToProviderResponse ValueOf(string message)
        {
            var packet = new GatewayToProviderResponse();
            packet.message = message;
            return packet;
        }
    }


    public class GatewayToProviderResponseRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 5001;
        }

        public void Write(ByteBuffer buffer, object packet)
        {
            if (packet == null)
            {
                buffer.WriteInt(0);
                return;
            }
            GatewayToProviderResponse message = (GatewayToProviderResponse) packet;
            buffer.WriteInt(-1);
            buffer.WriteString(message.message);
        }

        public object Read(ByteBuffer buffer)
        {
            int length = buffer.ReadInt();
            if (length == 0)
            {
                return null;
            }
            int beforeReadIndex = buffer.ReadOffset();
            GatewayToProviderResponse packet = new GatewayToProviderResponse();
            string result0 = buffer.ReadString();
            packet.message = result0;
            if (length > 0) {
                buffer.SetReadOffset(beforeReadIndex + length);
            }
            return packet;
        }
    }
}

File: Assets/zfoolua/gateway/GatewayToProviderResponse.lua

local GatewayToProviderResponse = {}

function GatewayToProviderResponse:new()
    local obj = {
        message = "" -- string
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function GatewayToProviderResponse:protocolId()
    return 5001
end

function GatewayToProviderResponse:protocolName()
    return GatewayToProviderResponse
end

function GatewayToProviderResponse:__tostring()
    local jsonTemplate = "{message:%s}"
    local result = string.format(jsonTemplate, self.message)
    return result
end

function GatewayToProviderResponse:write(buffer, packet)
    if packet == nil then
        buffer:writeInt(0)
        return
    end
    buffer:writeInt(-1)
    buffer:writeString(packet.message)
end

function GatewayToProviderResponse:read(buffer)
    local length = buffer:readInt()
    if length == 0 then
        return nil
    end
    local beforeReadIndex = buffer:getReadOffset()
    local packet = GatewayToProviderResponse:new()
    local result0 = buffer:readString()
    packet.message = result0
    if length > 0 then
        buffer:setReadOffset(beforeReadIndex + length)
    end
    return packet
end

return GatewayToProviderResponse
